src/components/Map.jsx

```diff
@@ -1,89 +1,75 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
 import {
-  withGoogleMap,
-  withScriptjs,
-  GoogleMap,
-  DirectionsRenderer,
+	withGoogleMap,
+	withScriptjs,
+	GoogleMap,
+	DirectionsRenderer,
 } from 'react-google-maps';
 
-class Map extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isLoading: true,
-      directions: null,
-    };
-  }
+/**
+ * The Map Component.
+ *
+ * @class Map
+ * @extends React.Component
+ * @param {Array<Object>} myPlaces
+ */
+const Map = ({ myPlaces }) => {
+	/** @constant {boolean} */
+	const [isLoading, setIsLoading] = useState(true);
+	/** @constant {Object} */
+	const [directions, setDirections] = useState(null);
```

Comment:
do we know internally typeof directions and setDirections?
Otherwise Array<Object> else Array<Object<additional type>, Object<additional typedef>>

Reply:
changed it using this thread -https://stackoverflow.com/questions/57976184/react-hooks-jsdoc-with-destructured-array-variables